specification/communication/Communication.Messages/models.tsp

```diff
@@ -205,6 +257,215 @@ model AudioNotificationContent extends NotificationContent {
+  @doc("Message notification type is Interactive.")
+  kind: CommunicationMessageKind.interactive;
+
+  @doc("The interactive message content.")
+  interactiveMessage: InteractiveMessage;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The Interactive message content to which user can read and respond.")
+model InteractiveMessage {
+  @doc("Gets or Sets Header content. Supports the following types:text, images etc.")
+  header?: MessageContent;
+
+  @doc("Gets or Sets Message body content. Emojis, markdown, and links are supported.")
+  body: TextMessageContent;
+
+  @doc("Gets or Sets Message footer content. Emojis, markdown, and links are supported.")
+  footer?: TextMessageContent;
+
+  @doc("The binding object to get or set Action which describes options user have to respond to message.")
+  action: ActionBindings;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@discriminator("kind")
+@doc("The message content object used to create interactive messages components.")
+model MessageContent {
+  @doc("Kind of MessageContent.")
+  kind: MessageContentKind;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type text information.")
+model TextMessageContent extends MessageContent {
+  @doc("Message content kind is text.")
+  kind: MessageContentKind.text;
+
+  @doc("The text value.")
+  text: string;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type document information.")
+model DocumentMessageContent extends MessageContent {
+  @doc("Message content kind is document.")
+  kind: MessageContentKind.document;
+
+  ...MediaMessageContent;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type image information.")
+model ImageMessageContent extends MessageContent {
+  @doc("Message content kind is image.")
+  kind: MessageContentKind.image;
+
+  ...MediaMessageContent;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type video information.")
+model VideoMessageContent extends MessageContent {
+  @doc("Message content kind is video.")
+  kind: MessageContentKind.video;
+
+  ...MediaMessageContent;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type Button information.")
+model ButtonContent {
+  @doc("Unique Id of the button content.")
+  id: string;
+
+  @doc("Title of the button content.")
+  title: string;
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type ButtonSet/ List of buttons information.")
+model ButtonSetContent extends MessageContent {
+  @doc("Message content kind is Button.")
+  kind: MessageContentKind.buttonSet;
+
+  @doc("Unique Id of the button content.")
+  buttons: ButtonContent[];
+}
+
+@added(Versions.c2024_11_15_Preview)
+@doc("The message content of type Url information.")
+model LinkContent extends MessageContent {
+  @doc("Message content kind is url.")
+  kind: MessageContentKind.url;
+
+  @doc("Title of the url content.")
+  title: string;
+
+  @clientName("uri", "csharp")
```

Comment:
All TCGC decorators (e.g. `@clientName`) should be added in client.tsp

Reply:
Moved the decorators added in the model tsp for interactive types.